app/jobs/process_digital_object_import_job.rb

```diff
@@ -1,82 +1,149 @@
 class ProcessDigitalObjectImportJob
   @queue = Hyacinth::Queue::DIGITAL_OBJECT_IMPORT
-  MAX_REQUEUES = 3
 
   def self.perform(digital_object_import_id)
     # Retrieve DigitalObjectImport instance from table
     digital_object_import = DigitalObjectImport.find(digital_object_import_id)
 
-    # User who created this job
-    user = digital_object_import.import_job.user
+    # If prerequisite check fails, return immediately.
+    # Re-queueing or mark-as-failure logic is handled by the prerequisite_row_check method.
+    return unless prerequisite_row_check(digital_object_import)
 
+    user = digital_object_import.import_job.user
     digital_object_data = JSON.parse(digital_object_import.digital_object_data)
 
-    if digital_object_data['pid'].present? && DigitalObject::Base.exists?(digital_object_data['pid'])
-      digital_object = existing_object_for_update(digital_object_data, user, digital_object_import)
+    if existing_object?(digital_object_data)
+      # We're updating data for an existing object
+      digital_object = existing_object_for_update(digital_object_data, user)
     else
-      digital_object = new_object(digital_object_data, user, digital_object_import)
+      digital_object = new_object(digital_object_data['digital_object_type']['string_key'], user, digital_object_import)
     end
 
-    status = assign_data(digital_object, digital_object_data, digital_object_import)
+    result = assign_data(digital_object, digital_object_data, digital_object_import)
 
-    return if status == :requeued
+    if result == :parent_not_found
```

Comment:
I wonder if we can retry more and wait less so that we aren't holding up the processing for 30sec if it ends up processing in 10sec. Have you done any testing around this? I don't think its a reason to hold up the pull request, but we might need to revisit later once this new processing is used in production.

Reply:
Good suggestion.  I'll do a loop in 10-second increments instead!